feat: orbit_get_original_request func to get Unified Endpoint Original Request Message

tokidoki11 opened the pull request:
## Description

Added a function for getting original request sent to Unified Endpoint via`int32_t soracom_get_original_request_as_string` function

reference: 
https://github.com/soracom/orbit-sdk-rust/pull/2


## Others
- deprecate soracom_release_userdata, use soracom_release_buffer instead

tokidoki11 committed:
feat: add original_data get orbit function

tokidoki11 commented on soracom/orbit.c:
```diff
@@ -146,15 +146,55 @@ int32_t soracom_get_userdata_as_string(const char** buf, size_t* siz) {
  * @param buf [in] The pointer to the allocated userdata.
  *
  * @note The pointer buf is allocated by soracom_get_userdata_as_string().
+ * @deprecated Use soracom_release_buffer instead
+ * 
```
is there any better way to set a function as deprecated?

tried: https://en.cppreference.com/w/c/language/attributes/deprecated

but there is no warning whatsoever on build. on vscode there is no strikethrough

bearmini replied:
Which version of emcc did you use? It seems that the `[[deprecated]]` attribute is supported after C23 and emcc ver >= 3.0.0 has support for C23 standard.

tokidoki11 replied:
my environment

```
sen@WorkMBP code % emcc --check
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.73-git
shared:INFO: (Emscripten: Running sanity checks)
sen@WorkMBP code % clang --version
Apple clang version 16.0.0 (clang-1600.0.26.4)
Target: arm64-apple-darwin24.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

bearmini replied:
I have added `[[deprecated]]` attribute and confirmed it works @ [cab5fb9](https://github.com/soracom/orbit-sdk-c/pull/2/commits/cab5fb9b3dd7c648308b3914766922e027da7f60)

tokidoki11 replied:
Thanks!

tokidoki11 replied:
Thanks!

I confirmed it works!

bearmini committed:
add [[deprecated]] attribute to soracom_release_userdata

bearmini commented:
Should we make `soracom_release_input_buffer` deprecated as well, maybe?

tokidoki11 committed:
add add [[deprecated]] attribute soracom_release_input_buffer

bearmini commented:
Anyway, which code formatter do you use for C/C++? Let's run it in our Makefile if possible.

tokidoki11 committed:
chore: add devcontainer

tokidoki11 commented:
@bearmini 

> Should we make soracom_release_input_buffer deprecated as well, maybe?

Yes, I think having one for all buffer free memory function would be easier to understand [34890d5](https://github.com/soracom/orbit-sdk-c/pull/2/commits/34890d5a4d852be00b1310d01479dca2e1a71462)

> Anyway, which code formatter do you use for C/C++? Let's run it in our Makefile if possible.

I currently dont run any formatter (which we should)
Im using vscode. I added devcontainer using emscripten [chore: add devcontainer](https://github.com/soracom/orbit-sdk-c/pull/2/commits/96d60201147691ecb210791abbcaf94ecd1a29fe)

This devcontainer and its vscode settings solves
1. no need to install emscripten on host machine
2. include emscripten that was marked as broken is now fixed (example code)

bearmini commented:
> > Anyway, which code formatter do you use for C/C++? Let's run it in our Makefile if possible.
> 
> I currently dont run any formatter (which we should) 

hmm, soracom/orbit.h was formatted on [34890d5](https://github.com/soracom/orbit-sdk-c/pull/2/commits/34890d5a4d852be00b1310d01479dca2e1a71462) without any semantic changes :thinking: 

vscode did it for you, or you did it by hand?

tokidoki11 committed:
revert formatting

tokidoki11 commented:
@bearmini 
oh sorry, i think vscode did it when i was playing with configuration settings....
I add changes to deprecate `soracom_release_input_buffer`

btw, reverted
[abbc3ed](https://github.com/soracom/orbit-sdk-c/pull/2/commits/abbc3edd82f686381a830490a0ac4a73d919c17e)

bearmini reviewed (approved):
LGTM :sparkles:
